discovery/discovery.go

```diff
@@ -202,14 +205,13 @@ type FailedAttempt struct {
 	Error  error
 }
 
-func walker(out *Endpoints, attempts *[]FailedAttempt, testFn DiscoverWalkFunc) DiscoverWalkFunc {
-	return func(pre string, eps *Endpoints, err error) error {
+func walker(attempts *[]FailedAttempt, testFn DiscoverWalkFunc) DiscoverWalkFunc {
+	return func(pre string, dd *discoveryData, err error) error {
 		if err != nil {
 			*attempts = append(*attempts, FailedAttempt{pre, err})
 			return nil
 		}
-		out.Append(*eps)
```

Comment:
Probably this was needed when there was a unique function for endpoints and keys discovery to walk up if a endpoint was found but not a keys and merge them. But I'm quite sure this led to some wrong endpoints/keys in the final result depending on the structure of the meta tags in the various paths.